controllers/csm_controller.go

```diff
@@ -163,6 +163,14 @@ func (r *ContainerStorageModuleReconciler) Reconcile(ctx context.Context, req ct
 		ConfigDirectory: r.Config.ConfigDirectory,
 	}
 
+	// perform prechecks
```

Comment:
Why are we changing the order? Is there a reason we want to do a precheck before checking if a file should be deleted?

One problem we will most likely face is when we deploy a CR which fails precheck.  If we try to delete that bad CR, the operator will prevent this as the precheck will  still be failing.  Nonetheless, I don't see why we need to run precheck for CR that is already scheduled for deletion.

I'll recommend we at least move this precheck to after checking if the CR is being deleted.

Reply:
Or we can leave owner ref check and driver/module pre-req checks in the Precheck function and call the precheck function after checking for deletion.

Reply:
We cannot call Prechecks after deletion. one of the reasons we are adding owner reference is to check if we have an existing installation of driver that was previously deployed by operator with owner reference. if we delete the driver even before checking for owner reference, it defeats the purpose and i wont be able to check owner reference if we delete the driver.
if we have a driver which fails other prechecks and doesn't have a owner reference then the driver has to be manually deleted.